dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/ingest/UningestInfoServiceTest.java

```diff
@@ -0,0 +1,84 @@
+package edu.harvard.iq.dataverse.ingest;
+
+import edu.harvard.iq.dataverse.DataFileServiceBean;
+import edu.harvard.iq.dataverse.persistence.MocksFactory;
+import edu.harvard.iq.dataverse.persistence.datafile.DataFile;
+import edu.harvard.iq.dataverse.persistence.datafile.DataTable;
+import edu.harvard.iq.dataverse.persistence.datafile.FileMetadata;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class UningestInfoServiceTest {
+
+    private DataFileServiceBean dataFileService = new DataFileServiceBean();
+    private UningestInfoService service = new UningestInfoService(dataFileService);
+
+
+    @ParameterizedTest
+    @CsvSource({
+            "true, text/xml-graphml, true, B, false",
+            "true, text/csv, false, A, true",
+            "true, text/csv, true, A, false",
+            "false, text/csv, true, A, true",
+            "false, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, true, D, false",
+            "false, text/tsv, true, A, true"
+    })
+    void listUningestableFiles_AND_hasUningestableFiles__variousFiles(boolean expectedEmpty, String mime, boolean singleVersion,
+                                             char ingestStatus, boolean hasDataTable) {
+        // given
+        Dataset dataset = prepareDatasetWithDataFile(mime, singleVersion, ingestStatus, hasDataTable ? new DataTable() : null);
+
+        // when
+        List<DataFile> files = service.listUningestableFiles(dataset);
+        boolean hasUningestableFiles = service.hasUningestableFiles(dataset);
+
+
+        // then
+        assertThat(files).hasSize(expectedEmpty ? 0 : 1);
+        assertThat(hasUningestableFiles).isEqualTo(!expectedEmpty);
+    }
+
+    @Test
+    void listUningestableFiles_AND_hasUningestableFiles__uinngestableDatasets() {
+        // given
+        Dataset releasedDataset = prepareDatasetWithDataFile("text/csv", true, 'D', null);
+        releasedDataset.getLatestVersion().setVersionState(DatasetVersion.VersionState.RELEASED);
+        Dataset noFilesDataset = new Dataset();
+
+        for (Dataset dataset : new Dataset[] { releasedDataset, noFilesDataset }) {
+            // when
+            List<DataFile> files = service.listUningestableFiles(dataset);
+            boolean hasUningestableFiles = service.hasUningestableFiles(dataset);
+
+            // then
+            assertThat(files).isEmpty();
+            assertThat(hasUningestableFiles).isFalse();
+        }
```

Comment:
IMO these should be two separate tests. When one of the case will fail - the only way to check which one is to run test on debug.